src/mistral_common/tokens/tokenizers/mistral.py

```diff
@@ -104,15 +103,37 @@ def v3(cls, is_tekken: bool = False, is_mm: bool = False) -> "MistralTokenizer":
         return cls.from_file(str(cls._data_path() / tokenizer_name), mode=ValidationMode.test)
 
     @classmethod
-    def v7(cls) -> "MistralTokenizer":
+    def v7(cls, is_mm: bool = False) -> "MistralTokenizer":
         """mistral-large 2.1"""
-        return cls.from_file(
-            str(cls._data_path() / "mistral_instruct_tokenizer_241114.model.v7m1"), mode=ValidationMode.test
-        )
+        if is_mm:
+            return cls.from_file(
+                str(cls._data_path() / "mistral_instruct_tokenizer_241114.model.v7m1"), mode=ValidationMode.test
+            )
+        else:
+            return cls.from_file(
+                str(cls._data_path() / "mistral_instruct_tokenizer_241114.model.v7"), mode=ValidationMode.test
+            )
 
     @classmethod
-    def from_model(cls, model: str) -> "MistralTokenizer":
+    def from_model(cls, model: str, strict: bool = False) -> "MistralTokenizer":
         model_name_to_tokenizer_cls: Dict[str, Callable[[], MistralTokenizer]] = {
+            "ministral-8b-2410": lambda: MistralTokenizer.v3(is_tekken=True),
+            "mistral-tiny-2312": MistralTokenizer.v2,
+            "open-mistral-nemo-2407": lambda: MistralTokenizer.v3(is_tekken=True),
+            "mistral-tiny-2407": MistralTokenizer.v3,
+            "mistral-small-2312": MistralTokenizer.v2,
+            "open-mixtral-8x22b-2404": MistralTokenizer.v3,
+            "mistral-small-2402": MistralTokenizer.v2,
+            "mistral-small-2409": lambda: MistralTokenizer.v3(is_tekken=True),
+            "mistral-medium-2312": MistralTokenizer.v1,
+            "mistral-large-2402": MistralTokenizer.v2,
+            "mistral-large-2407": MistralTokenizer.v3,
+            "mistral-large-2411": MistralTokenizer.v7,
+            "pixtral-large-2411": lambda: MistralTokenizer.v7(is_mm=True),
+            "codestral-2405": MistralTokenizer.v3,
+            "codestral-mamba-2407": MistralTokenizer.v3,
+            "pixtral-12b-2409": lambda: MistralTokenizer.v3(is_tekken=True, is_mm=True),
+            # The following are deprecated - only left for backward comp. Delete in >= 1.6.0
```

Comment:
All of the following names do not have a date and thus cannot be tracked correctly. They will be removed in mistral_common >= 1.6.0